pkg/expr/sql/db.go

```diff
@@ -17,10 +29,342 @@ func (db *DB) RunCommands(commands []string) (string, error) {
+	for _, frame := range frames {
+		keyed[frame.RefID] = append(keyed[frame.RefID], frame)
+	}
+	return keyed
+}
+
+func normalizeSchemas(frames []*data.Frame) {
+	fields := map[string]*data.Field{}
+	for _, f := range frames {
+		for _, fld := range f.Fields {
+			fields[fld.Name] = fld
+		}
+	}
+	for _, fld := range fields {
+		for _, f := range frames {
+			found := false
+			for _, fld2 := range f.Fields {
+				if fld2.Name == fld.Name {
+					found = true
+					break
+				}
+			}
+			if !found {
+				makeArray := maker[fld.Type()]
+				arr := makeArray(f.Rows())
+				nullField := data.NewField(fld.Name, fld.Labels, arr)
+				f.Fields = append(f.Fields, nullField)
+			}
+		}
+	}
+}
+
+var maker = map[data.FieldType]func(length int) any{
+	data.FieldTypeFloat64:         func(length int) any { return makeArray[float64](length) },
+	data.FieldTypeFloat32:         func(length int) any { return makeArray[float32](length) },
+	data.FieldTypeInt16:           func(length int) any { return makeArray[int16](length) },
+	data.FieldTypeInt64:           func(length int) any { return makeArray[int64](length) },
+	data.FieldTypeInt8:            func(length int) any { return makeArray[int8](length) },
+	data.FieldTypeUint8:           func(length int) any { return makeArray[uint8](length) },
+	data.FieldTypeUint16:          func(length int) any { return makeArray[uint16](length) },
+	data.FieldTypeUint32:          func(length int) any { return makeArray[uint32](length) },
+	data.FieldTypeUint64:          func(length int) any { return makeArray[uint64](length) },
+	data.FieldTypeNullableFloat64: func(length int) any { return makeArray[*float64](length) },
+	data.FieldTypeNullableFloat32: func(length int) any { return makeArray[*float32](length) },
+	data.FieldTypeNullableInt16:   func(length int) any { return makeArray[*int16](length) },
+	data.FieldTypeNullableInt64:   func(length int) any { return makeArray[*int64](length) },
+	data.FieldTypeNullableInt8:    func(length int) any { return makeArray[*int8](length) },
+	data.FieldTypeNullableUint8:   func(length int) any { return makeArray[*uint8](length) },
+	data.FieldTypeNullableUint16:  func(length int) any { return makeArray[*uint16](length) },
+	data.FieldTypeNullableUint32:  func(length int) any { return makeArray[*uint32](length) },
+	data.FieldTypeNullableUint64:  func(length int) any { return makeArray[*uint64](length) },
+	data.FieldTypeString:          func(length int) any { return makeArray[string](length) },
+	data.FieldTypeNullableString:  func(length int) any { return makeArray[*string](length) },
+	data.FieldTypeTime:            func(length int) any { return makeArray[time.Time](length) },
+	data.FieldTypeNullableTime:    func(length int) any { return makeArray[*time.Time](length) },
+}
+
+func makeArray[T any](length int) []T {
+	return make([]T, length)
+}
+
+func (db *DB) QueryFramesInto(tableName string, query string, frames []*data.Frame, f *data.Frame) error {
```

Comment:
I think it would be good to break this up some. One part of loading for frames (that takes the frames grouped by RefId), and returns (something - db instance or connection), and then another func for querying that.